src/Swarm/Game/Step.hs

```diff
@@ -645,6 +645,8 @@ execConst c vs k = do
           (`isJustOr` cmdExn View [ "There is no robot named ", s, " to view." ])
 
         lift . lift $ viewCenterRule .= VCRobot s
+        lift . lift $ focusedRobotName .= s
```

Comment:
Another solution would be to use a helper function:

```haskell
setFocusedRobot :: Text -> GameState -> GameState
setFocusedRobot name = (focusedRobotName .~ name) . (viewCenterRule .~ VCRobot name)
```

to replace those two updates with `lift . lift $ modify (setFocusedRobot s)`